File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        s = str(x)
        
        left = 0
        right = len(s)-1
        
        while left <= right:
            
            if s[left] == s[right]:
                left += 1
                right -= 1
            
            else:
                return False
        
        return True

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        if(x < 0)
        {
            return false;
        }
        
        int original = x;
        int reversed = 0;
            

        while (x != 0)
        {
            reversed = (reversed * 10) + (x % 10);
            x = x / 10;
        }
        

        if (original == reversed)
            return true;
        else
            return false;
        
    }
}